lib/dialects/oracle/utils.js

```diff
@@ -93,6 +93,7 @@ function isConnectionError(err) {
     'ORA-56600', // an illegal OCI function call was issued
     'NJS-024',
     'NJS-003',
+    'NJS-500',
```

Comment:
Maybe we can include NJS-521 as well, because I'm receiving this error together with NJS-500:

```
select * from table(***.***(:1, :2)) - NJS-500: connection to the Oracle Database was broken

NJS-521: connection to host *** port ***received end-of-file on communication channel. (CONNECTION_ID=***)
```

Reply:
Thanks for reviewing.

Only the start of `err.message` is looked up here (see line 98 below). Hence, NJS-521, as it is not in the beginning of the string, would never be matched here.

That said, I didn't find other missing `NJS-` error code that should be added here, this [search](https://github.com/search?q=repo%3Aoracle/node-oracledb%20errors.getErr&type=code) lead me to think there are none missing.

Reply:
Yes, never mind. Check [this](https://github.com/oracle/node-oracledb/blob/faa59030b5a5aee62f203f67fee9b5e0e930026d/lib/thin/sqlnet/ntTcp.js#L327), the error will always be wrapped with NJS-500, so you can check if NJS-500 is present

Reply:
Reviewed: adding the NJS-500 is a big improvement to the current situation, and so far no one has reported other valid codes to add.  Thank you!